hyperspy/defaults_parser.py

```diff
@@ -56,15 +56,15 @@ def guess_gos_path():
 
 if os.path.isfile(defaults_file):
     # Remove config file if obsolated
-    f = open(defaults_file)
-    if 'Not really' in f.readline():
-        # It is the old config file
-        f.close()
-        _logger.info('Removing obsoleted config file')
-        os.remove(defaults_file)
-        defaults_file_exists = False
-    else:
-        defaults_file_exists = True
+    with open(defaults_file) as f:
```

Comment:
Done


On 21/07/16 16:00, Tomas Ostasevicius wrote:
>
> In hyperspy/defaults_parser.py 
> <https://github.com/hyperspy/hyperspy/pull/1181#discussion_r71722515>:
>
> > @@ -56,15 +56,15 @@ def guess_gos_path():
> >
> >  if os.path.isfile(defaults_file):
> >      # Remove config file if obsolated
> > -    f = open(defaults_file)
> > -    if 'Not really' in f.readline():
> > -        # It is the old config file
> > -        f.close()
> > -        _logger.info('Removing obsoleted config file')
> > -        os.remove(defaults_file)
> > -        defaults_file_exists = False
> > -    else:
> > -        defaults_file_exists = True
> > +    with open(defaults_file) as f:
>
> ...it seems strange to use |with| statement and then close and even 
> delete the file inside the scope.
>
> Could you disentangle the two steps?
>
> i.e.
>
> with  open(defaults_file)as  f:
>      if  'Not really'  in  f.readline():
>         defaults_file_exists=  False
>      else:
>         defaults_file_exists=  True
> if  not  defaults_file_exists:
>      _logger.info('Removing obsoleted config file')
>      os.remove(defaults_file)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/hyperspy/hyperspy/pull/1181/files/ffd84157f89df532ebc7ef7f698ce36d7ad412fc#r71722515>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AA8aF5VMBF2VQ48SXZWq-wefIC7e1BHxks5qX4mRgaJpZM4JQsGs>.
>